### DISCUSSION TITLE
Multiple Windows

### DISCUSSION THREAD

**Discussion by Tyemccahon** [TomSchimansky/CustomTkinter#371] [Q&A]
hay how can you have more then one window, so one opens and the other one closes,
so i can make a login screen that closes and opens a new screen

**Comment by JEdmario16** [comment]
Hello! You can use TopLevel class

like this:

```import tkinter as tk
from tkinter import ttk
import customtkinter


app = customtkinter.CTk()
app.geometry("400x200")
btn = customtkinter.CTkButton(master=app, text="click-me to open a new window", command=lambda: create_new_window(app))
btn.pack()


def create_new_window(master):
    window = customtkinter.CTkToplevel(master=master)
    window.geometry("400x200")
    window.title("New window")

    label = customtkinter.CTkLabel(master=window, text="This is another window")
    label.pack()


app.mainloop()
```

When you click the button, create_new_window() is used as callback function and use app as master. 
if you want some example with OO, this tutorial can help you: https://github.com/TomSchimansky/CustomTkinter/wiki/CTkToplevel

**Reply by Observer55** [reply]
Thanks for the example code, but I have a question too. When we create a toplevel window, and then when we close the main window, all the toplevel windows are closing. That's OK normally, but how can we make it, so it can stay when we the main windows close.
(Sorry for my English if I have some grammatik problems...)

**Comment by Adarsh-365** [comment]
```
from tkinter import ttk
import customtkinter


app = customtkinter.CTk()
app.geometry("400x200")
btn = customtkinter.CTkButton(master=app, text="click-me to open a new window", command=lambda: create_new_window(app))
btn.pack()

def create_new_window(master):

    window  = customtkinter.CTk()
    window.geometry("400x200")
    window.title("New window")
    btn = customtkinter.CTkButton(master=window, text="click-me to open a new window", command=lambda: create_new_window(app))
    btn.pack()

    label = customtkinter.CTkLabel(master=window, text="This is another window")
    label.pack()
    window.mainloop()


app.mainloop()

```


this will work, but having two mainloop is not good practise

**Reply by Anonymous6598** [reply]
Use `CTkTopLevel` instead of `CTk`. Therefore you won't need `.mainloop()` method.
